authentication/logout/service/logout_service.go

```diff
@@ -72,7 +73,14 @@ func (s *logoutServiceImpl) Logout(ctx context.Context, redirectURL string) (str
 	parameters.Add("redirect_uri", redirectURL)
 	logoutURL.RawQuery = parameters.Encode()
 
-	// TODO implement logout business logic here, i.e. invalidate all of the user's tokens
+	// If a token string was passed, then set the status to "logged out" for all tokens with the same identity
+	if tokenString != nil {
```

Comment:
Well, you actually could extract the token from the context here instead of the controller. And if there is a token then "log out" all the tokens. Then you could call this service from both controller endpoints logout and logoutv2. So, you don't have to check if the token is present twice (in controller v2 and in the service).